packages/cloud-cognitive/src/components/InlineEdit/_inline-edit.scss

```diff
@@ -18,24 +18,196 @@
 // InlineEdit uses the following Carbon for IBM Products components:
 // TODO: @import(s) of IBM Products component styles used by InlineEdit
 
+// cut down version borrowed from 'carbon-components/scss/globals/scss/vendor/@carbon/styles/scss/utilities/focus-outline';
+@mixin input-outline($color: $focus) {
+  outline: $spacing-01 solid $color;
+  outline-offset: calc(-1 * $spacing-01);
+
+  @media screen and (prefers-contrast) {
+    outline-style: dotted;
+  }
+}
+
+@mixin input-button-defaults() {
+  width: var(--size);
+  height: 100%;
+  min-height: initial;
+  max-height: var(--size);
+
+  // stylelint-disable-next-line carbon/layout-token-use
+  padding: calc(0.5 * (var(--size) - var(--icon-size)));
+  border: 2px solid transparent;
+}
+
 // Define all component styles in a mixin which is then exported using
 // the Carbon import-once mechanism.
 @mixin inline-edit {
   // The block part of our conventional BEM class names (blockClass__E--M).
   $block-class: #{$pkg-prefix}--inline-edit;
 
-  .#{$block-class} {
+  @include block-wrap($block-class, 3) {
```

Comment:
Woah there horsey...  I like the ingenuity here, but this feels like a big hammer. Depth 3? This is going to bloat the CSS considerably, with three copies of the whole block class on every CSS rule. I'm guessing you were fighting a specificity war, but one problem with this approach is that it passes the whole battle on to our users: if anyone wants to override any of the CSS styles, they are going to have to out-specify us, which means all those rules will need specificity 4 or more. It's going to be very hard to override any of our styles!

I think we should be a lot more selective -- if there are some rules that need extra specificity, we can add repeated classes there, rather than swamping our whole CSS in a three-level-deep specificity wrapper. One advantage of popping `&.#{$block-class}.#{$block-class} .#{$block-class}--entry-field {` on each affected rule is that it's a clear and maintainable declaration that we needed to boost the specificity here.